### PR TITLE
HP-1671:  new prop `averege_power_consumption`

### PR DESCRIPTION
**Pull request by tafid**

### CODE REVIEW

File: src/models/DeviceProperties.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        return [
+            'set-properties' => 'set-properties',
+        ];
+    }
+
+    public function rules(): array
+    {
+        return [
+            [['id'], 'integer', 'on' => ['set-properties']],
+            [['average_power_consumption'], 'string', 'on' => ['set-properties']],
```
Comment (SilverFire) [inline src/models/DeviceProperties.php:27]:
```suggestion
            [['average_power_consumption'], 'number', 'on' => ['set-properties']],
```

File: src/views/hub/view.php [hunk 2]
```diff
@@ -123,6 +123,10 @@
                         'digit_capacity_id',
                         'nic_media',
                         'base_port_no',
+                        [
+                            'attribute' => 'deviceProperties.average_power_consumption',
+                            'label' => 'Average Power Consumption',
```
Comment (SilverFire) [inline src/views/hub/view.php:128]: i18n is missing

File: src/views/server/modal/hardwareSettings.php [hunk 3]
```diff
@@ -3,32 +3,52 @@
...
         <div class="col-md-12">
-            <?php $box = Box::begin(['renderBody' => false, 'options' => ['class' => 'box-widget'], 'bodyOptions' => ['class' => 'no-padding']]) ?>
+            <?php $form = ActiveForm::begin([
+                'action' => Url::to(['@device/set-properties']),
+                'id' => 'apc-form',
+                'options' => ['style' => ['margin-bottom' => '1em']],
+            ]) ?>
+            <?= Html::activeHiddenInput($model->deviceProperties, 'id') ?>
+            <?= $form->field($model->deviceProperties, 'average_power_consumption') ?>
+            <?= Html::submitButton('Save', ['class' => 'btn btn-success', 'style' => 'display: inline-block']) ?>
```
Comment (SilverFire) [inline src/views/server/modal/hardwareSettings.php:28]: i18n is missing

**Review by SilverFire — CHANGES_REQUESTED** [review]

File: src/models/DeviceProperties.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        return [
+            [['id'], 'integer', 'on' => ['set-properties']],
+            [['average_power_consumption'], 'number', 'on' => ['set-properties']],
+        ];
+    }
+
+    public function attributeLabels(): array
+    {
+        return [
+            'average_power_consumption' => Yii::t('hipanel:server', 'Average Power Consumption'),
```
[the remark this answers is not in the document]
Comment (tafid) [reply src/models/DeviceProperties.php:35]: Обсудили на дейли

**Review by SilverFire — APPROVED** [review]

**Pull request merged**